infrastructure/terraform/components/reporting/scripts/create_iceberg_table.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Creates Iceberg table for reporting if it doesn't exist already
+
+ENV=${1:-"no_env"}
+
+if [[ ${ENV} == "no_env" ]]; then
+    echo "Environment name not provided as first argument"
+    exit 1
+fi
+
+table_exists=$(aws glue get-tables --database-name nhs-notify-${ENV}-reporting-database | jq 'any(.TableList[].Name == "'nhs_notify_${ENV}_item_status_iceberg'"; .)')
+
+if [[ ${table_exists} == "true" ]]; then
+    echo "Table already exists for this environment in the database, hence exiting gracefully"
+    exit 0
+fi
+
+
+account_id=""
+
+if [[ ${ENV} == "prod" ]]; then
```

Comment:
Doesn't seem right that the account ids are embedded in conditional logic in the script. Shouldn't this be data driven and come from variables?

Reply:
thank you, thats a great spot, I was trying it out locally before and forgot, now it will be passed as the 2nd arg :)